### PR TITLE
RSDK-9294 -  add extra to captureAll response

### PR DESCRIPTION
**Pull request by Rob1in**

### CODE REVIEW

File: services/vision/client_test.go [hunk 1]
```diff
@@ -142,14 +143,18 @@ func TestClient(t *testing.T) {
...
 		opts := viscapture.CaptureOptions{
 			true,
-			true, true, true,
+			true,
+			true,
+			true,
 		}
-		capt, err := client.CaptureAllFromCamera(context.Background(), "", opts, map[string]interface{}{})
+		extra := map[string]interface{}{"foo": "captureAll"}
+		capt, err := client.CaptureAllFromCamera(context.Background(), "", opts, extra)
```
Comment (bhaney) [inline services/vision/client_test.go:151]: Can you also add a test to see what happens if you pass "nil" in the extra field?

File: services/vision/server_test.go [hunk 2]
```diff
@@ -170,12 +171,14 @@ func TestServerCaptureAllFromCamera(t *testing.T) {
...
 			Detections: []objectdetection.Detection{det1},
+			Extra:      extra,
 		}, nil
 	}
 	captAllResp, err := server.CaptureAllFromCamera(context.Background(), &captureRequest)
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, len(captAllResp.Detections), test.ShouldEqual, 1)
-	test.That(t, getDetectionsResp.Detections[0].GetClassName(), test.ShouldEqual, "yes")
+	test.That(t, captAllResp.Detections[0].GetClassName(), test.ShouldEqual, "yes")
+	test.That(t, captAllResp.Extra.AsMap(), test.ShouldResemble, map[string]interface{}{"foo": "GetDetections"})
```
Comment (bhaney) [inline services/vision/server_test.go:181]: Also here, can you create a test to show that you don't get a fatal error if nil is passed in extra

**Review by bhaney — CHANGES_REQUESTED** [review]
Nice quick fix -- just add two tests for testing when extra is nil

**Comment by bhaney** [comment]
Another thing you'll want to do is implicitly call the DoCommand in the CaptureAllFromCamera method in line 392 in services/vision/vision.go

File: services/vision/client_test.go (hunk 1 above)
Comment (Rob1in) [reply services/vision/client_test.go:151]: ok, done!

File: services/vision/server_test.go (hunk 2 above)
Comment (Rob1in) [reply services/vision/server_test.go:182]: done
